api_v3/services/BatchControlService.php

```diff
@@ -170,6 +184,38 @@ private function getOrCreateScheduler(KalturaScheduler $scheduler)
 		
 		return $schedulerDb;
 	}
+
+
+	/**
+	 * batch getOrCreateScheduler returns a scheduler by name, create it if doesn't exist
+	 *
+	 * @param KalturaScheduler $scheduler
+	 * @return Scheduler
+	 */
+	private function getOrCreateSchedulerByHostAndName(KalturaScheduler $scheduler)
+	{
+		$c = new Criteria();
+		$c->add ( SchedulerPeer::CONFIGURED_ID, null , criteria::ISNOTNULL);
+		$c->add ( SchedulerPeer::HOST, $scheduler->host);
+		$c->add ( SchedulerPeer::NAME, $scheduler->name);
+		$schedulerDb = SchedulerPeer::doSelectOne($c, myDbHelper::getConnection(myDbHelper::DB_HELPER_CONN_PROPEL2));
+
+		if($schedulerDb)
+		{
+			return $schedulerDb;
+		}
+
+		$schedulerDb = new Scheduler();
+		$schedulerDb->setLastStatus(time());
+		$schedulerDb->setName($scheduler->name);
+		$schedulerDb->setHost($scheduler->host);
+		$schedulerDb->setDescription('');
+		$schedulerDb->save();
+
+		$schedulerDb->setConfiguredId($schedulerDb->getId() + self::SCHEDULER_BASE_ID);
```

Comment:
What if this configured ID already exist? must not override existing one.

Reply:
see above - if you got to that point it means you are creating a new one.